api/src/pcapi/core/users/email/update.py

```diff
@@ -209,7 +209,7 @@ def cancel_email_update_request(encoded_token: str) -> None:
     user = models.User.query.get(token.user_id)
     if not user:
         raise exceptions.InvalidToken()
-    new_email = token.data["new_email"]
+    new_email = token.data["new_email"] if "new_email" in token.data else None
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🤓 nitpick:** 
```suggestion
    new_email = token.data.get("new_email")

‏